### PR TITLE
TILA-1635: Allow reservations without opening hours if flag is enabled

### PR DESCRIPTION
**Pull request by saaste**
The PR adds a new boolean flag to ReservationUnit. If the flag is set true, reservations can be done without opening hours.

### CODE REVIEW

File: api/graphql/tests/test_reservations/test_reservation_create.py [hunk 1]
```diff
@@ -406,6 +406,32 @@ def test_create_fails_when_reservation_unit_closed_on_selected_time(
...
+        response = self.query(self.get_create_query(), input_data=input_data)
+        content = json.loads(response.content)
+
+        assert_that(content.get("errors")).is_none()
+        assert_that(
+            content.get("data").get("createReservation").get("errors")
+        ).does_not_contain(
+            "Reservation unit is not open within desired reservation time."
+        )
+
```
Comment (nicobav) [inline api/graphql/tests/test_reservations/test_reservation_create.py:434]: We should add here a check that the actual reservation has been created.

File: api/graphql/reservations/reservation_serializers.py [hunk 2]
```diff
@@ -157,6 +157,37 @@ def validate_reservee_type(self, value):
             )
         return value
 
+    def validate_reservation(
```
[the remark this answers is not in the document]
Comment (saaste) [reply api/graphql/reservations/reservation_serializers.py:160]: Sure, I can do some refactoring. I also discussed with @kieferbonk that with the flag enabled we should also skip the interval check. I mean the one that gives `Reservation start time does not match the allowed interval of 60 minutes.` error. I'll add that too.

File: api/graphql/tests/test_reservations/test_reservation_create.py (hunk 1 above)
Comment (saaste) [reply api/graphql/tests/test_reservations/test_reservation_create.py:436]: Added! 👍

File: api/graphql/tests/test_reservations/test_reservation_update.py [hunk 3]
```diff
@@ -315,6 +315,31 @@ def test_update_fails_when_reservation_unit_closed_on_selected_time(
...
+        response = self.query(self.get_update_query(), input_data=input_data)
+        content = json.loads(response.content)
+
+        assert_that(content.get("errors")).is_none()
+        assert_that(
+            content.get("data").get("updateReservation").get("errors")
+        ).does_not_contain(
+            "Reservation unit is not open within desired reservation time."
+        )
+
```
[the remark this answers is not in the document]
Comment (saaste) [reply api/graphql/tests/test_reservations/test_reservation_update.py:346]: Added! 👍

**Comment by saaste** [comment]
@nicobav I updated the commit message. Now it describes how serializer is updated 👍

**Review by nicobav — APPROVED** [review]
Nice! Special niceyness for the refactoring of the validate stuff in its own methods :raised_hands:

**Pull request merged**